plenum/test/restart/test_network_i3pc_watcher.py

```diff
@@ -0,0 +1,101 @@
+import pytest
+
+from plenum.server.i3pc_watchers import NetworkI3PCWatcher
+
+DEFAULT_NODE_SET = {'Alpha', 'Beta', 'Gamma', 'Delta'}
+
+
+class WatcherCallbackMock:
+    def __init__(self):
+        self.call_count = 0
+
+    def __call__(self):
+        self.call_count += 1
+
+
+@pytest.fixture
+def watcher():
+    cb = WatcherCallbackMock()
+    watcher = NetworkI3PCWatcher(cb)
+    watcher.set_nodes(DEFAULT_NODE_SET)
+    return watcher
+
+
+def _add_node(watcher: NetworkI3PCWatcher, node: str):
+    nodes = watcher.nodes
+    nodes.add(node)
+    watcher.set_nodes(nodes)
```

Comment:
At first I was thinking that watchdog could potentially react to change in number of nodes, that's why I'm using setter (which is also called from production code) instead of directly manipulating field. Also I'm a little bit afraid of adding _add_node/_remove_node to `Watcher` class because someone might "optimize" them to change nodes directly instead of going through `set_nodes`, which will render tests useless.

I think renaming `nodes` to `_nodes` and adding read-only property getter can make intentions more clear, what do you think?